### PR TITLE
Make Cruise Control send alert when execution is slow.

### PR DESCRIPTION
**Pull request by kidkun**
Fix issue #673 

Also this patch will mark the user request as `CompletedWithError`(originally it was `Completed`) in `UserTaskManger` and returned in `/user_tasks` response for requests whose execution is stopped(by user or CC itself) or interrupted.

### CODE REVIEW

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/ExecutionTask.java [hunk 1]
```diff
@@ -197,6 +203,20 @@ public void completed(long now) {
     _endTime = now;
   }
 
+  /**
+   * Check whether the execution time of the task is too long.
+   */
+  public boolean shouldReportExecutionTooSlow(long now) {
```
Comment (efeg) [inline cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/ExecutionTask.java]: Looks like after reporting this once, we always want to mute. Does it make sense to move (1) the logic regarding the reporting functionality, and (2) muting the task inside this method? -- i.e. a change from `shouldReportExecutionTooSlow` to `reportExecutionTooSlow`?

File: cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/executor/ReplicationThrottleHelperTest.java [hunk 2]
```diff
@@ -61,7 +61,7 @@ private void createTopics() {
   }
 
   private ExecutionTask inProgressTaskForProposal(long id, ExecutionProposal proposal) {
-    ExecutionTask task = new ExecutionTask(id, proposal, ExecutionTask.TaskType.INTER_BROKER_REPLICA_ACTION);
+    ExecutionTask task = new ExecutionTask(id, proposal, ExecutionTask.TaskType.INTER_BROKER_REPLICA_ACTION, 0);
```
Comment (efeg) [inline cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/executor/ReplicationThrottleHelperTest.java]: `maxExecutionTime` of 0 seems confusing. Should we ensure that there is a sanity check in `ExecutionTask` that ensures a positive value for `maxExecutionTime`?

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/ExecutionTaskPlanner.java [hunk 3]
```diff
@@ -218,7 +221,8 @@ private void maybeAddIntraBrokerReplicaMovementTasks(Collection<ExecutionProposa
           String currentLogdir = currentLogdirByReplica.get(new TopicPartitionReplica(proposal.topic(), proposal.partitionId(), r.brokerId()));
           if (currentLogdir != null && !currentLogdir.equals(r.logdir())) {
             long replicaActionExecutionId = _executionId++;
-            ExecutionTask task = new ExecutionTask(replicaActionExecutionId, proposal, r.brokerId(), INTRA_BROKER_REPLICA_ACTION);
+            ExecutionTask task = new ExecutionTask(replicaActionExecutionId, proposal, r.brokerId(), INTRA_BROKER_REPLICA_ACTION,
+                                                   Math.round(proposal.dataToMoveInMB() / MIN_REPLICA_MOVEMENT_RATE));
```
[the remark this answers is not in the document]
Comment (kidkun) [reply cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/ExecutionTaskPlanner.java]: good idea!

File: cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/executor/ReplicationThrottleHelperTest.java [hunk 4]
```diff
@@ -61,7 +61,7 @@ private void createTopics() {
   }
 
   private ExecutionTask inProgressTaskForProposal(long id, ExecutionProposal proposal) {
-    ExecutionTask task = new ExecutionTask(id, proposal, ExecutionTask.TaskType.INTER_BROKER_REPLICA_ACTION);
+    ExecutionTask task = new ExecutionTask(id, proposal, ExecutionTask.TaskType.INTER_BROKER_REPLICA_ACTION, 100);
```
Comment (efeg) [inline cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/executor/ReplicationThrottleHelperTest.java]: Nit: Let's avoid hard coding.

**Pull request merged**